server/src/main/java/timely/api/model/Metric.java

```diff
@@ -34,10 +34,23 @@
 @WebSocket(operation = "put")
 public class Metric implements TcpRequest, HttpPostRequest, WebSocketRequest {
 
+    private static final ThreadLocal<ByteBuffer> WRITE_BUFFER = new ThreadLocal<ByteBuffer>() {
```

Comment:
Not being completely familiar with the rest of the code, is this class being used in a thread pool or are additional threads being created as need be?  If the latter, then perhaps a ByteBuffer pool might server you better.